README-cmake-details.md

```diff
@@ -0,0 +1,223 @@
+
+```
+|-- components/
+|   |-- rust-test/
+|       |-- CMakeLists.txt
+|       |-- placeholder.c
+|       ^^^ These two files are standard ESP-IDF CMake component build files, with CMakeLists.txt being slightly more complicated than usual.
+|       |
+|       |-- build.rs
+|       |-- Cargo.toml
+|       |-- rust-toolchain.toml
+|       |-- src/
+|           |-- lib.rs
+|       ^^^ All these other files are what typically constitutes the layout of a Rust library crate and they are NOT specifc to ESP-IDF.
+|           You can google any Rust source on the internet as to their purpose and meaning.
+```
+
+### ESP-IDF Component Portion
+
+#### `components/rust-test/CMakeLists.txt`:
+
+This file builds the `/components/rust-test` ESP-IDF component. It is however more complex than usual, as it needs to setup quite a few things so that calling into Rust `cargo` is running smoothly. The content of this file is the main boilerplate that the `esp-idf-template` automates.
+
+You can of course modify this file post-generation. The various variables and overall behavior of the file is documented with inline comments.
+
+#### `components/rust-test/placeholder.c`:
+
+As mentioned previously, this is an empty C file which is only there so as the CMake->Cargo integration magic to work.
+
+### Rust Library Crate Portion
+
+#### `components/rust-test/build.rs`:
+
+This file is a standard Rust [build scriptlet](https://doc.rust-lang.org/cargo/reference/build-scripts.html). It would be empty if you have not opted into the "HAL" option (note that by default "HAL" is enabled).
+
+If you have selected the "HAL" option, the `build.rs` file will contain a call into the `embuild` library that nakes sure that your component can properly link against - and in general - can "talk" to the [Safe Rust bindings for ESP-IDF](https://github.com/esp-rs/esp-idf-svc) (the "HAL").
+
+#### `components/rust-test/Cargo.toml`:
+
+This is a pretty standard and mostly empty `cargo` [build manifest file](https://doc.rust-lang.org/cargo/reference/manifest.html), which sets up your Rust crate to be built as a Rust static libray. 
+
+If you have selected the "HAL" option when generating the project, the manifest would also list:
+* a dependency to `esp-idf-svc` (i.e. [the safe Rust wrappers for ESP-IDF](https://github.com/esp-rs/esp-idf-svc));
+* a built-time dependency to [`embuild`](https://github.com/esp-rs/embuild) which helps the build integration between ESP-IDF and Rust to run smoothly);
+* the popular Rust [`log`](https://github.com/rust-lang/log) crate.
+
+Post-generation, you can add/remove additional dependencies on Rust crates here as well as modify how Rust compiles your project (as in optimization level and others).
+
+#### `components/rust-test/rust-toolchain.toml`:
```

Comment:
Maybe link to https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file?